File: L2-Sorting/QuickSort/main.js
function quick(arr,low,high){
	if(low>high){
		return;
	}
	let i=low-1;
	let j=low;
	while(j<high){
		if(arr[j]<arr[high]){
			i++;
			let tmp = arr[j];
			arr[j]=arr[i];
			arr[i]=tmp;
		}
		j++;
	}
	let tmp=arr[high];
	arr[high]=arr[i+1];
	arr[i+1]=tmp;
	quick(arr,low,i);
	quick(arr,i+2,high);
}


let arr = [5, 10, 2, 3, 0, 100, 30];
console.log("INITIAL: ["+arr+"]");
quick(arr,0,arr.length-1);
console.log("SORTED: ["+arr+"]");

File: L2-Sorting/QuickSort/main.py
def quick(arr,low,high):
	if(low>=high):
		return
	i=low-1
	j=low
	while j<high:
		if arr[j]<arr[high]:
			i+=1
			tmp = arr[i]
			arr[i]=arr[j]
			arr[j]=tmp
		j+=1
	tmp = arr[high]
	arr[high]=arr[i+1]
	arr[i+1]=tmp
	quick(arr,low,i)
	quick(arr,i+2,high)

arr = [5,10,2,3,0,100,30]
print("INITIAL: "+str(arr))
quick(arr,0,len(arr)-1)
print("FINAL: "+str(arr))
